### PR TITLE
Enable email-alert-service to update subscriber list titles

### PR DESCRIPTION
**Pull request by huwd**
[Trello](https://trello.com/c/IkYegsOt/1225-update-subscriberlist-titles-on-major-minor-publishing-change)

### CODE REVIEW

**Comment by huwd** [comment]
Sorry this shoulda been a draft, do not merge!

File: modules/govuk/manifests/apps/email_alert_service.pp [hunk 1]
```diff
@@ -31,6 +31,11 @@
 #   Whether or not to enable the unpublishing worker
 #   Default: false
 #
+#
```
Comment (barrucadu) [inline modules/govuk/manifests/apps/email_alert_service.pp:34]: Extra unnecessary blank line

File: modules/govuk/manifests/apps/email_alert_service.pp [hunk 2]
```diff
@@ -65,6 +71,20 @@
...
   }
 
+  govuk::procfile::worker { 'email-alert-service-subscriber-list-details-update-major-consumer':
+    setenv_as      => $app_name,
+    enable_service => $enable_subscriber_list_update_queue_consumers,
+    process_type   => 'subscriber-list-details-update-major-consumer',
+    process_regex  => '\/rake message_queues:subscriber_list_details_update_major_consumer',
+  }
+
+  govuk::procfile::worker { 'email-alert-service-subscriber-list-details-update-major-consumer':
```
Comment (barrucadu) [inline modules/govuk/manifests/apps/email_alert_service.pp:81]: Should be `email-alert-service-subscriber-list-details-update-minor-consumer`

File: modules/govuk/manifests/apps/email_alert_service.pp [hunk 3]
```diff
@@ -65,6 +71,20 @@
...
+    setenv_as      => $app_name,
+    enable_service => $enable_subscriber_list_update_queue_consumers,
+    process_type   => 'subscriber-list-details-update-minor-consumer',
+    process_regex  => '\/rake message_queues:subscriber_list_details_update_major_consumer',
```
Comment (barrucadu) [inline modules/govuk/manifests/apps/email_alert_service.pp:85]: Should be `minor` here too

File: modules/govuk/manifests/apps/email_alert_service.pp (hunk 2 above)
Comment (huwd) [reply modules/govuk/manifests/apps/email_alert_service.pp:81]: urgh that got lost in a rebase, sorry 1 sec

File: modules/govuk/manifests/apps/email_alert_service.pp [hunk 4]
```diff
@@ -65,6 +70,20 @@
...
+    setenv_as      => $app_name,
+    enable_service => $enable_subscriber_list_update_queue_consumers,
+    process_type   => 'subscriber-list-details-update-minor-consumer',
+    process_regex  => '\/rake message_queues:subscriber_list_details_update_minor_consumer',
+  }
+
+  govuk::procfile::worker { 'email-alert-service-subscriber-list-details-update-major-consumer':
+    setenv_as      => $app_name,
+    enable_service => $enable_subscriber_list_update_queue_consumers,
+    process_type   => 'subscriber-list-details-update-minor-consumer',
```
Comment (barrucadu) [inline modules/govuk/manifests/apps/email_alert_service.pp:83]: Now this `process_type` says `minor` but the rest of it is for the `major`

File: modules/govuk/manifests/apps/email_alert_service/rabbitmq.pp [hunk 5]
```diff
@@ -82,8 +117,8 @@
   govuk_rabbitmq::consumer { $amqp_user:
     ensure               => $ensure,
     amqp_pass            => $amqp_pass,
-    read_permission      => "^(amq\\.gen.*|${amqp_major_change_queue}|${amqp_unpublishing_queue}|${amqp_exchange})\$",
-    write_permission     => "^(amq\\.gen.*|${amqp_major_change_queue}|${amqp_unpublishing_queue})\$",
-    configure_permission => "^(amq\\.gen.*|${amqp_major_change_queue}|${amqp_unpublishing_queue})\$",
+    read_permission      => "^(amq\\.gen.*|${amqp_major_change_queue}|${amqp_unpublishing_queue}|${ampq_subscriber_list_update_minor_queue}|${ampq_subscriber_list_update_major_queue}${amqp_exchange})\$",
```
Comment (huwd) [inline modules/govuk/manifests/apps/email_alert_service/rabbitmq.pp:120]: we didn't talk about these in our catchup, but looking at the others I think this is needed too?

File: modules/govuk/manifests/apps/email_alert_service.pp [hunk 6]
```diff
@@ -40,6 +44,7 @@
   $redis_host = undef,
   $email_alert_api_bearer_token = undef,
   $enable_unpublishing_queue_consumer = false,
+  $enable_subscriber_list_update_queue_consumers =  false,
```
Comment (barrucadu) [inline modules/govuk/manifests/apps/email_alert_service.pp:47]: Extra unnecessary space before the `false` (kind of surprised the linter doesn't complain about that, it's really picky about puppet spacing elsewhere...)

File: modules/govuk/manifests/apps/email_alert_service.pp (hunk 4 above)
Comment (huwd) [reply modules/govuk/manifests/apps/email_alert_service.pp]: yep gimme 1 sec, i'll hit the "re-review" button in a sec

File: modules/govuk/manifests/apps/email_alert_service/rabbitmq.pp (hunk 5 above)
Comment (barrucadu) [reply modules/govuk/manifests/apps/email_alert_service/rabbitmq.pp:120]: Yes, it gives the email-alert-service rabbitmq user access to those queues

**Review by barrucadu — APPROVED** [review]

**Pull request merged**